cmd/core/main.go

```diff
@@ -93,8 +94,18 @@ func main() {
 	flag.IntVar(&controllerArgs.RevisionLimit, "revision-limit", 50,
 		"RevisionLimit is the maximum number of revisions that will be maintained. The default value is 50.")
 	flag.StringVar(&healthAddr, "health-addr", ":9440", "The address the health endpoint binds to.")
-	flag.BoolVar(&controllerArgs.ApplyOnceOnly, "apply-once-only", false,
+
+	flag.BoolVar(&applyOnceOnly, "apply-once-only", false,
 		"For the purpose of some production environment that workload or trait should not be affected if no spec change")
+	flag.BoolVar(&applyOnceOnlyForce, "apply-once-only-force", false,
+		"Stronger case for applyOnceOnly, it will override the value of flag 'ApplyOnceOnly' ")
+	controllerArgs.ApplyMode = oamcontroller.ApplyOnceOnlyOff
+	if applyOnceOnlyForce {
+		controllerArgs.ApplyMode = oamcontroller.ApplyOnceOnlyForce
+	} else if applyOnceOnly {
+		controllerArgs.ApplyMode = oamcontroller.ApplyOnceOnlyOn
+	}
+
```

Comment:
To keep backward compatible, apply-once-only flag is retained and a new but similar flag apply-once-only-force is added. If it's okay to remove apply-once-only boolean flag, I would like to just use one string flag(off/on/force) instead of two.

Reply:
yes, use `off/on/force` and update the design doc.

Reply:
you can also make the compatibility:

apply-once-only is a string

`apply-once-only="true" or "on"  ` means enable it.
`apply-once-only="" or "false" or "off"  ` means disable it.
`apply-once-only="force"  means enable force apply only once.

In other case, please report an error on start.

Reply:
Got it. Good idead!

Reply:
yes, true/false in yaml can be easily convert to string